app/controllers/admin/interests_controller.rb

```diff
@@ -0,0 +1,62 @@
+class Admin::InterestsController < ApplicationController
+  before_action :authenticate_user!
+
+  def show
+    interest = Interest.find(params[:id])
+    authorize interest
+    render :show, locals: { interest: interest }
+  end
+
+  def index
+    interests = Interest.all
+    authorize interests
+    render :index, locals: { interests: interests }
+  end
+
+  def new
+    interest = Interest.new
+    authorize interest
+    render :new, locals: { interest: interest }
+  end
+
+  def create
+    interest = Interest.new(interest_params)
+    authorize interest
+
+    if interest.save
+      redirect_to admin_interest_path(interest)
+    else
+      render :new
+    end
+  end
+
+  def edit
+    interest = Interest.find(params[:id])
+    authorize interest
+    render :edit, locals: { interest: interest }
+  end
+
+  def update
+    interest = Interest.find(params[:id])
+    authorize interest
+
+    if interest.update_attributes(interest_params)
+      redirect_to admin_interest_path
```

Comment:
does that redirect to a single updated interest or raises an error (no route sth)?